### ISSUE TITLE
Relative volume mounts on Windows fail based on the casing of the current directory in the shell

### ISSUE THREAD

**Issue by jklawrence** [docker/compose#7173]
## Description of the issue
Relative volume mounts fail based on the casing of the current directory in the shell. 

## Context information (for bug reports)
I've only been able to reproduce this in PowerShell, as CMD Prompt always seems to use the "correct" casing of the folder. But if I have a folder eg "C:/Foo" on my filesystem and I browse to "C:/foo" in PowerShell and attempt to mount a volume using a relative file path I get an "error while creating mount source path". If I browse to "C:/Foo" and run the same command, it works.

I believe it's an issue in docker-compose because I can't reproduce this with just `docker run` and the `-v` flag.

I've confirmed with a co-worker that this works fine on docker-compose 1.24.1

**Output of `docker-compose version`**
```
docker-compose version 1.25.2, build 698e2846
docker-py version: 4.1.0
CPython version: 2.7.16
OpenSSL version: OpenSSL 1.0.2q  20 Nov 2018
```

**Output of `docker version`**
```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:37 2019
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
services:
  alpine:
    image: alpine
    volumes:
    - C:\foo\bar:/bar:rw
version: '3.7'
```

## Steps to reproduce the issue

1. Open PowerShell and run the following commands:
```
cd c:/
mkdir Foo
cd foo
mkdir bar
```
2. Add the following docker-compose.yml to the "C:/Foo" directory:
```
version: '3.7'
services:
  alpine:
    image: alpine
    volumes:
      - ./bar:/bar
```
3. Execute `docker-compose up`

### Observed result
Error creating mount source path.

### Expected result
Container starts up. If you cd from "C:/foo" to "C:/Foo" and run `docker-compose up` again, this is the result you get.

### Stacktrace / full error message

```
ERROR: for c0b28291010e_foo_alpine_1  Cannot start service alpine: error while creating mount source path '/host_mnt/c/foo/bar': mkdir /host_mnt/c/foo: file exists

ERROR: for alpine  Cannot start service alpine: error while creating mount source path '/host_mnt/c/foo/bar': mkdir /host_mnt/c/foo: file exists
```

## Additional information

Windows 10 Enterprise
Version 1809 (OS Build 17763.914)

Docker-Compose installed via Docker for Windows CE.

**Comment by zippos** [comment]
Same issue here after update

**Comment by herkrath** [comment]
Same issue here as well after updating to Docker Desktop 2.2.0.0.

**Comment by abrose** [comment]
Had the same issue today after updating Docker on Windows to 19.03.5. I'm using Docker on Windows 10 in combination with WSL. I'm using a bash alias in order to cd into my project directory. After battling the issue for hours I found this post. Then I started examining every possible usage of paths in my setup. Finally I realized that in my alias I've been using the path "/c/users/<myusername>/Documents/projectdir" instead of "/c/Users/..." with a capital U. After changing this, restarting docker and rebuilding all images from scratch my containers finally start again.

Al
